bigtable-dataflow-import/src/test/java/com/google/cloud/bigtable/dataflowimport/ImportIntegrationTest.java

```diff
@@ -0,0 +1,265 @@
+
+  @Before
+  public void setup() {
+    commonOptions = PipelineOptionsFactory.as(HBaseImportOptions.class);
+    String projectId =
+        checkNotNull(System.getProperty(BigtableOptionsFactory.PROJECT_ID_KEY), "Project Id");
+    commonOptions.setBigtableProjectId(projectId);
+    commonOptions.setBigtableClusterId(
+        checkNotNull(System.getProperty(BigtableOptionsFactory.CLUSTER_KEY), "Cluster Id"));
+    commonOptions.setBigtableZoneId(
+        checkNotNull(System.getProperty(BigtableOptionsFactory.ZONE_KEY), "Zone Id"));
+    commonOptions.setStagingLocation(
+        checkNotNull(System.getProperty("google.dataflow.stagingLocation"), "Staging Location"));
+
+    cloudTestDataFolder =
+        checkNotNull(System.getProperty(CLOUD_TEST_DATA_FOLDER), CLOUD_TEST_DATA_FOLDER);
+    if (!cloudTestDataFolder.endsWith(File.separator)) {
+      cloudTestDataFolder = cloudTestDataFolder + File.separator;
+    }
+  }
+
+  private HBaseImportOptions createImportOptions(
+      String importFilePath, String tableName, boolean isHBase094Format) {
+    HBaseImportOptions importOptions = commonOptions.as(HBaseImportOptions.class);
+    importOptions.setFilePattern(importFilePath);
+    importOptions.setBigtableTableId(tableName);
+    importOptions.setHBase094DataFormat(isHBase094Format);
+    if (importFilePath.startsWith("gs://")) {
+      importOptions.setRunner(BlockingDataflowPipelineRunner.class);
+    } else {
+      importOptions.setRunner(DirectPipelineRunner.class);
+    }
+    return importOptions;
+  }
+
+  @Test
+  public void testImportSequenceFile_onLocalFs() throws Exception {
+    String tableId = "test_" + UUID.randomUUID().toString();
+    File sequenceFile =tmpFolder.newFile(tableId);
+    Set<? extends Cell> expected = createTestData();
+    SequenceFileIoUtils.createFileWithData(sequenceFile, expected);
+
+    HBaseImportOptions options = createImportOptions(
+        sequenceFile.getPath(), tableId, false /** isHBase094Format **/);
+    BigtableTableUtils tableAdmin = BigtableTableUtils.from(options);
+    tableAdmin.createTable(Bytes.toString(CF));
+
+    doImport(options);
+
+    assertCellSetsEqual(expected, tableAdmin.readAllCellsFromTable());
+    tableAdmin.dropTable();
+  }
+
+  /**
+   * Verifies that a Sequence File on Google Cloud Storage (GCS) can be imported
+   * into Google Cloud Bigtable.
+   *
+   * <p>This method creates a local Sequence File, copies it to the stagingLocation folder
+   * on GCS, then imports the GCS file. The GCS file is not removed by post-test cleanup.
+   */
+  @Test
+  public void testImportSequenceFile_onGcs() throws Exception {
+    runImportFileTestOnGcs(false /** isHBase094Format **/);
+  }
+
+  @Test
+  public void testImportSequenceFile_onGcsWithHBase094Format() throws Exception {
+    runImportFileTestOnGcs(true /** isHBase094Format **/);
+  }
+
+  private void runImportFileTestOnGcs(boolean isHBase094Format) throws Exception {
```

Comment:
Cases 1 and 4 are sharing code. 
Case 2 throws, while case 3 returns normally but nothing would be imported.
I tried to extract common code for 2 and 3 but the parameter list gets longer.